taichi/backends/cuda/cuda_device.cpp

```diff
@@ -31,6 +31,28 @@ DeviceAllocation CudaDevice::allocate_memory(const AllocParams &params) {
   return alloc;
 }
 
+DeviceAllocation CudaDevice::allocate_memory_runtime(const AllocParams &params,
+                                                     JITModule *runtime_jit,
+                                                     LLVMRuntime *runtime,
+                                                     uint64 *result_buffer) {
+  AllocInfo info;
+  if (params.host_read || params.host_write) {
```

Comment:
Can we make more code into the shared impl? IMO except L39 here, `allocate_memory_runtime` in `CpuDevice` and `CudaDevice` are still the same

Reply:
I have thought about it. First, line 39 as well as line 46 are GPU specific. Second, if we move the allocation code, the other allocate_memory function should be changed as well. Maybe there can be another PR later to just add something like a LLVMDevice layer. WDYT?

Reply:
Sounds good.